src/historywork/ResolveSnapshotWork.cpp

```diff
@@ -16,11 +16,21 @@ ResolveSnapshotWork::ResolveSnapshotWork(
     , mSnapshot(snapshot)
     , mTimer(std::make_unique<VirtualTimer>(app.getClock()))
 {
+    if (!mSnapshot)
+    {
+        throw std::runtime_error("ResolveSnapshotWork: invalid snapshot");
+    }
 }
 
 BasicWork::State
 ResolveSnapshotWork::onRun()
 {
+    if (mSnapshot->mLocalState.futuresAllClear())
```

Comment:
why do we need to special case the "all clear" case?

Reply:
Looks like `futuresAllResolved` already accounts for "all clear", so I think we don't need it. Removed.